src/main/scala/scalaz/nio/channels/GatheringByteChannel.scala

```diff
@@ -1,20 +1,44 @@
 package scalaz.nio.channels
 
+import java.nio.{ ByteBuffer => JByteBuffer }
 import java.nio.channels.{ GatheringByteChannel => JGatheringByteChannel }
 
-import scalaz.nio.ByteBuffer
-import scalaz.zio.IO
+import scalaz._
+import Scalaz._
+import scalaz.nio.{ Buffer }
+import scalaz.zio.{ Chunk, IO }
+import scalaz.zio.interop.scalaz72._
 
 class GatheringByteChannel(private val channel: JGatheringByteChannel) {
 
-  def write(srcs: Seq[ByteBuffer], offset: Int, length: Int): IO[Exception, Long] =
+  final private[nio] def writeBuffer(
+    srcs: IList[Buffer[Byte]],
+    offset: Int,
+    length: Int
+  ): IO[Exception, Long] =
     IO.syncException(channel.write(unwrap(srcs), offset, length))
 
-  def write(srcs: Seq[ByteBuffer]): IO[Exception, Long] =
+  final def write(srcs: IList[Chunk[Byte]], offset: Int, length: Int): IO[Exception, Long] =
```

Comment:
Why this double wrapping `IList[Chunk[Byte]]` instead of just `Chunk[Byte]`?

Reply:
Definition of GatheringByteChannel is `A gathering write operation writes, in a single invocation, a sequence of bytes from one or more of a given sequence of buffers.` IMHO If it doesn't take `List` of `Chunk`s it is no different from combinators in `AsynchronousByteChannel`.

Reply:
All right, makes sense! 

I think we should remove dependency on Scalaz and depend only on ZIO. Which means switching from `IList` to `List` and using `IO.collectAll` instead of `.sequence`.

Reply:
make sense!